### PR TITLE
JAMES-2920 No longer rely on mocks in ElasticSearchListeningMessageSe…

### PR DESCRIPTION
**Pull request by Arsnael**
…archIndexTest

### CODE REVIEW

File: mailbox/elasticsearch/src/test/java/org/apache/james/mailbox/elasticsearch/events/ElasticSearchListeningMessageSearchIndexTest.java [hunk 1]
```diff
@@ -19,81 +19,160 @@
...
+        .attachment(Attachment.builder()
+            .bytes("".getBytes(StandardCharsets.UTF_8))
+            .type("type")
+            .build())
+        .name("name")
+        .isInline(false)
+        .build();
+
+    private static final SimpleMailboxMessage MESSAGE_WITH_ATTACHMENT = MESSAGE_BUILDER.messageId(MESSAGE_ID_1)
+        .uid(MESSAGE_UID_1)
```
Comment (rouazana) [inline mailbox/elasticsearch/src/test/java/org/apache/james/mailbox/elasticsearch/events/ElasticSearchListeningMessageSearchIndexTest.java]: you should not use different messageId and different uid than other message in this fixture?

File: mailbox/elasticsearch/src/test/java/org/apache/james/mailbox/elasticsearch/events/ElasticSearchListeningMessageSearchIndexTest.java [hunk 2]
```diff
@@ -19,81 +19,160 @@
...
-        MailboxSessionMapperFactory mapperFactory = mock(MailboxSessionMapperFactory.class);
-        messageToElasticSearchJson = mock(MessageToElasticSearchJson.class);
-        ElasticSearchSearcher elasticSearchSearcher = mock(ElasticSearchSearcher.class);
-        SessionProvider mockSessionProvider = mock(SessionProvider.class);
+    public void setup() throws MailboxException, IOException {
+        mapperFactory = new InMemoryMailboxSessionMapperFactory();
+
+        MessageToElasticSearchJson messageToElasticSearchJson = new MessageToElasticSearchJson(
+            new DefaultTextExtractor(),
+            ZoneId.of("Europe/Paris"),
```
Comment (rouazana) [inline mailbox/elasticsearch/src/test/java/org/apache/james/mailbox/elasticsearch/events/ElasticSearchListeningMessageSearchIndexTest.java]: not UTC?

File: mailbox/elasticsearch/src/test/java/org/apache/james/mailbox/elasticsearch/events/ElasticSearchListeningMessageSearchIndexTest.java [hunk 3]
```diff
@@ -103,168 +182,234 @@ public void deserializeElasticSearchListeningMessageSearchIndexGroup() throws Ex
...
-        BulkResponse expectedBulkResponse = mock(BulkResponse.class);
-        when(elasticSearchIndexer.delete(any(List.class)))
-            .thenReturn(Optional.of(expectedBulkResponse));
-
-        //When
-        testee.delete(session, mailbox, Lists.newArrayList(MESSAGE_UID));
-
-        //Then
-        verify(elasticSearchIndexer).delete(eq(Lists.newArrayList(ELASTIC_SEARCH_ID)));
+    public void deleteShoulRemoveIndex() throws IOException {
```
Comment (rouazana) [inline mailbox/elasticsearch/src/test/java/org/apache/james/mailbox/elasticsearch/events/ElasticSearchListeningMessageSearchIndexTest.java]: Should

File: mailbox/elasticsearch/src/test/java/org/apache/james/mailbox/elasticsearch/events/ElasticSearchListeningMessageSearchIndexTest.java [hunk 4]
```diff
@@ -103,168 +182,234 @@ public void deserializeElasticSearchListeningMessageSearchIndexGroup() throws Ex
...
+
+        elasticSearch.awaitForElasticSearch();
+
+        testee.delete(session, mailbox, Lists.newArrayList(MESSAGE_UID_1, MESSAGE_UID_2));
+        elasticSearch.awaitForElasticSearch();
+
+        SearchQuery query = new SearchQuery(SearchQuery.all());
+        assertThat(testee.search(session, mailbox, query))
+            .isEmpty();
+    }
```
Comment (rouazana) [inline mailbox/elasticsearch/src/test/java/org/apache/james/mailbox/elasticsearch/events/ElasticSearchListeningMessageSearchIndexTest.java]: do you have a delete test case where you don't remove all messages?

File: mailbox/elasticsearch/src/test/java/org/apache/james/mailbox/elasticsearch/events/ElasticSearchListeningMessageSearchIndexTest.java [hunk 5]
```diff
@@ -103,168 +182,234 @@ public void deserializeElasticSearchListeningMessageSearchIndexGroup() throws Ex
...
+        SearchQuery query = new SearchQuery(SearchQuery.flagIsSet(Flags.Flag.ANSWERED));
+        assertThat(testee.search(session, mailbox, query))
+            .containsExactly(MESSAGE_1.getUid());
     }
 
     @Test
-    public void updateShouldPropagateExceptionWhenExceptionOccurs() throws Exception {
-        //Given
-        Flags flags = new Flags();
+    public void updateShouldNotUpdateNorThrowOnUnknownMessageUid() throws Exception {
```
Comment (rouazana) [inline mailbox/elasticsearch/src/test/java/org/apache/james/mailbox/elasticsearch/events/ElasticSearchListeningMessageSearchIndexTest.java]: why?

Comment (rouazana) [reply mailbox/elasticsearch/src/test/java/org/apache/james/mailbox/elasticsearch/events/ElasticSearchListeningMessageSearchIndexTest.java]: and what about delete then?

Comment (Arsnael) [reply mailbox/elasticsearch/src/test/java/org/apache/james/mailbox/elasticsearch/events/ElasticSearchListeningMessageSearchIndexTest.java]: That's the behavior of update in the code. We first search if the message exists with its UID, if it does we update, if it doesn't we do nothing. Thus no exception.

I thought that `deleteShouldBeIdempotent` was proving implicitly that case but I can add an extra test just for this yes !

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**